### ISSUE TITLE
[descendants] Use provided index for SSR stability

### ISSUE THREAD

**Issue by travigd** [reach/reach-ui#624]
## 🐛 Bug report
I'm not sure if this qualifies as a bug, but seems like something that should be doable.

### Current Behavior

tl;dr: on initial render, `@reach/accordion` always passes `hidden={false}` to the `AccordionPanel` component, even if it _should_ be visible (e.g., because it's the first child in a non-collapsible accordion).

### Expected behavior

Reach should always pass the correct value of `hidden`.

### Reproducible example

Will add if requested.

### Suggested solution(s)

Currently, the issue is that the accordion uses the `useDescendant` hook, which always returns `index: -1` initially. It would be nice if we could have some way to make this resolve synchronously for the first render, even if it requires manually annotating indices.

### Additional context

I'm trying to animate the pop-in using Framer Motion. The issue is that the component is initially rendered in the `hidden` variant, and then changes to `visible`, which causes Framer to apply the animation. This results in the items that should be visible by default to animate in on page load (which I'm willing to deal with, but is sub-ideal).

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    |      `@reach/accordion`   |    `0.10.4`     |
| React            |         |         |
| Browser          |         |         |
| Assistive tech   |         |         |
| Node             |         |         |
| npm/yarn         |         |         |
| Operating System |         |         |

Related: #472 and https://github.com/reach/reach-ui/commit/e93835686cec6cf4c25e679629cbd6e4e4ceb838

**Comment by chancestrickland** [comment]
Yeah, we need an SSR bail out but that will likel requires an explicit index prop for uncontrolled components. Thanks for raising the issue.

**Comment by travigd** [comment]
In my own codebase, I was able to workaround this by adding an `initialStateHint` prop to my components.
Not exactly _pretty_ but it works for now. :^)

```js
const AnimatedAccordionPanel = ({ children, initialStateHint }) => {
  const { index, isExpanded } = useAccordionItemContext();

  // If index is -1, then Reach is still trying to figure out what the index is
  const isOpen = index === -1 && initialStateHint === "open" ? true : isExpanded;

  // Render with the initial animation state corresponding to isOpen
  // HideableDiv takes a prop isVisible and animates based on changes to that value
  return <AccordionPanel as={HideableDiv} isVisible={isOpen}>...</AccordionPanel>;
};
```

**Comment by chaance** [comment]
As of 0.16.1 this should work as expected. Pass an `index` to `AccordionItem and it will be available on the first render.
